workspaces/mcp-chat/plugins/mcp-chat-backend/src/services/MCPClientServiceImpl.ts

```diff
@@ -0,0 +1,471 @@
+      );
+
+      const baseServerConfig = {
+        id: serverConfig.id,
+        name: serverConfig.name,
+        type: serverConfig.type,
+        url: serverConfig?.url,
+        npxCommand: serverConfig?.npxCommand,
+        scriptPath: serverConfig?.scriptPath,
+        args: serverConfig?.args,
+      };
+
+      try {
+        const client = new Client({
+          name: `${serverConfig.name}-client`,
+          version: '1.0.0',
+        });
+
+        let transport;
+
+        if (serverConfig.type === 'streamable-http') {
+          // Streamable HTTP connection
+          if (!serverConfig.url) {
+            throw new Error(
+              `Server config for '${serverConfig.name}' with streamable-http type must have a url`,
+            );
+          }
+
+          const transportOptions: any = {};
+
+          // Add headers if provided
+          if (serverConfig.headers) {
+            transportOptions.requestInit = {
+              headers: serverConfig.headers,
+            };
+          }
+
+          transport = new StreamableHTTPClientTransport(
+            new URL(serverConfig.url),
+            transportOptions,
+          );
+        } else if (serverConfig.type === 'sse') {
+          // SSE connection
+          if (!serverConfig.url) {
+            throw new Error(
+              `Server config for '${serverConfig.name}' with SSE type must have a url`,
+            );
+          }
+
+          const transportOptions: any = { url: new URL(serverConfig.url) };
+
+          // Add headers if provided
+          if (serverConfig.headers) {
+            transportOptions.headers = serverConfig.headers;
+          }
+
+          transport = new SSEClientTransport(transportOptions);
+        } else {
+          // STDIO connection (default)
+          let command: string;
+          let args: string[];
+
+          if (serverConfig.npxCommand) {
+            // Use npm command - find npx executable
+            try {
+              command = await findNpxPath();
+              args = [
+                '-y',
+                serverConfig.npxCommand,
+                ...(serverConfig.args || []),
+              ];
+            } catch (error) {
+              throw new Error(
+                `Failed to find npx for server '${serverConfig.name}': ${
+                  error instanceof Error ? error.message : error
+                }. Please ensure Node.js is properly installed with npx available.`,
+              );
+            }
+          } else if (serverConfig.scriptPath) {
+            // Use script path
+            const isPythonScript = serverConfig.scriptPath.endsWith('.py');
+            const isWindows = process.platform === 'win32';
+
+            if (isPythonScript) {
+              command = isWindows ? 'python' : 'python3';
+            } else {
+              command = process.execPath;
+            }
+            args = [serverConfig.scriptPath, ...(serverConfig.args || [])];
```

Comment:
I think there is no documentation around this?

I also not sure if this magic is needed or if you can just people configure:

```yaml
  - id: local-script-server
    name: Local Script Server
    command:
      - python3
      - ./servers/my-mcp-server.js
      - --port=9100
      - --env=dev
```

So this isn't wrong. I just couldn't find that in the READMEs or SERVER_CONFIGURATION.md